src/v/kafka/client/direct_consumer/fetcher.cc

```diff
@@ -718,6 +598,207 @@ fetcher::process_fetch_response(
     co_return result;
 }
 
+ss::future<std::optional<fetched_partition_data>>
+fetcher::process_partition_response(
+  const model::topic& topic,
+  partition_data partition_response,
+  const topic_partition_map<epoch_set>& epochs,
+  fetch_response_content& result,
+  chunked_hash_map<model::topic, absl::flat_hash_set<model::partition_id>>&
+    dirty_partitions) {
+    vlog(
+      logger().trace,
+      "[broker: {}] topic: {}, partition fetch response: {}",
+      _id,
+      topic,
+      partition_response);
+
+    // the response will be consumed, grab some vars for logs
+    const auto partition_id = partition_response.partition_index;
+
+    // pull the records from the response if present
+    chunked_vector<model::record_batch> response_records{};
+    size_t response_size{0};
+    if (
+      partition_response.records.has_value()
+      && !partition_response.records->is_end_of_stream()) {
+        response_size = partition_response.records->size_bytes();
+        response_records = co_await reader_to_chunked_vector(
+          std::move(partition_response.records.value()));
+    }
+
+    // null this out as the records are already consumed
+    partition_response.records = std::nullopt;
+    if (response_size == 0) {
+        vassert(
+          response_records.size() == 0, "zero size should have zero records");
+    } else {
+        vassert(
+          response_records.size() != 0,
+          "nonzero size should have nonzero records");
+    }
+
+    fetched_partition_data part_data{};
+    part_data.error = partition_response.error_code;
+    part_data.partition_id = partition_response.partition_index;
+
+    auto maybe_epoch_set = find_epoch_set(
+      topic, partition_response.partition_index, epochs);
+
+    vassert(
+      maybe_epoch_set.has_value(),
+      "tp should be found in snapshotted epochs if the response is "
+      "epoch consistent");
```

Comment:
if the assertion fires, would you expect that someone would be able to debug with only this static string?